pkg/apis/clusterapi/validation/cluster_test.go

```diff
@@ -0,0 +1,153 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"bytes"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	clusteroperatorapi "github.com/openshift/cluster-operator/pkg/api"
+	clusteroperator "github.com/openshift/cluster-operator/pkg/apis/clusteroperator/v1alpha1"
+	jsonserializer "k8s.io/apimachinery/pkg/runtime/serializer/json"
+	"k8s.io/apiserver/pkg/endpoints/request"
+	"sigs.k8s.io/cluster-api/pkg/apis/cluster"
+)
+
+func getValidClusterProviderConfig() *clusteroperator.ClusterProviderConfigSpec {
+	spec := &clusteroperator.ClusterProviderConfigSpec{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ClusterProviderConfigSpec",
+			APIVersion: "clusteroperator.openshift.io/v1alpha1",
+		},
+		ClusterSpec: clusteroperator.ClusterSpec{
+			ClusterVersionRef: clusteroperator.ClusterVersionReference{
+				Name: "someclusterref",
+			},
+			MachineSets: []clusteroperator.ClusterMachineSet{
+				{
+					ShortName: "master",
+					MachineSetConfig: clusteroperator.MachineSetConfig{
+						NodeType: "Master",
+						Size:     1,
+					},
+				},
+				{
+					ShortName: "infra",
+					MachineSetConfig: clusteroperator.MachineSetConfig{
+						NodeType: "Compute",
+						Infra:    true,
+						Size:     1,
+					},
+				},
+			},
+		},
+	}
+	return spec
+}
+
+func getValidClusterWithEmptyProviderConfig() *cluster.Cluster {
+	cluster := &cluster.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-cluster",
+		},
+		Spec: cluster.ClusterSpec{
+			ProviderConfig: cluster.ProviderConfig{
+				Value: nil,
+			},
+		},
+	}
+	return cluster
+}
+
+func getRawExtensionFromProviderConfig(providerConfig *clusteroperator.ClusterProviderConfigSpec) *runtime.RawExtension {
+	serializer := jsonserializer.NewSerializer(jsonserializer.DefaultMetaFactory, clusteroperatorapi.Scheme, clusteroperatorapi.Scheme, false)
+	var buffer bytes.Buffer
+	serializer.Encode(providerConfig, &buffer)
+
+	return &runtime.RawExtension{
+		Raw: buffer.Bytes(),
+	}
+}
+
+func getValidCluster() *cluster.Cluster {
+	cluster := getValidClusterWithEmptyProviderConfig()
+	providerConfig := getValidClusterProviderConfig()
+
+	cluster.Spec.ProviderConfig.Value = getRawExtensionFromProviderConfig(providerConfig)
+
+	return cluster
+}
+
+func getValidClusterStrategy() *ClusterStrategy {
+	return &ClusterStrategy{}
+}
+
+// TestValidateCluster tests the Validate function.
+func TestValidateCluster(t *testing.T) {
+	cases := []struct {
+		name            string
+		clusterStrategy *ClusterStrategy
```

Comment:
Take this out of the cases unless you are planning to have a subtest that uses an invalid ClusterStrategy--which doesn't make too much sense anyway since the ClusterStrategy is the component under test. If you are trying to have different ClusterStrategy behavior based on the contained StorageBuilder, then include the StorageBuilder in the cases, instead of the ClusterStrategy.